tex/polyglossia.sty

```diff
@@ -1244,33 +1234,41 @@
 }
 
 \newcommand\xpg@ifdefined[3]{%
-    \ifluatex
-        \xpg@set@language@luatex@iv{#1}%
-    \fi
+    \xpg@set@hyphenation@patterns{#1}%
     \polyglossia@check@ifdefined:NTF{#1}{#2}{#3}
 }%
 
-\newcommand\xpg@set@language@luatex@iv[1]{%
+\newcommand*\xpg@set@bbl@hyphendata[1]{%
+    % here we use lu@texhyphen@loaded@\the\language, the same as in babel
+    \ifcsdef{bbl@hyphendata@#1}{}{%
+        \global\@namedef{bbl@hyphendata@\the\language}{}%
+    }
+}
+
+\newcommand\xpg@set@hyphenation@patterns[1]{%
```

Comment:
Add a comment about his function

Reply:
Thanks, I have addressed all concerns (I think). As to \xpg@set@hyphendata, I am not entirely sure myself why/when we need this. I hop @reutenauer can clarify this, then I can update the comment.